eng/pipelines/release-cli.yml

```diff
@@ -156,6 +156,8 @@ stages:
               AZD_TEST_TENANT_ID: $(arm-tenant-id)
               AZD_TEST_AZURE_SUBSCRIPTION_ID: $(SubscriptionId)
               AZD_TEST_AZURE_LOCATION: eastus2
+              AZD_TEST_REMOTE_STATE_ACCOUNT: saazdremotestate
+              AZD_TEST_REMOTE_STATE_CONTAINER: remote-env-test
```

Comment:
We've avoided static resources in favor of standing up resources per-run and tearing them down. 

Since the storage account lives outside the scope of the resources created for the test run we should create a separate resource group and storage account within that group and then tear those down. We do have a [script](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/New-TestResources.ps1) which does that for the Azure SDK Engineering System, but I'm not sure how coupled it is to our `sdk/<service>` directory layout and other engsys features (like environment variable export).

Reply:
I agree on this strategy.  I will do a fast follow PR to this to figure out how we want to handle these types of scenarios.